src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -830,6 +831,7 @@ private void StepInternal(int threadId, enum_STEPKIND stepKind, SteppingGranular
             }
             try
             {
+                ((IDebugReversibleEngineProgram160)m_engine).SetExecuteDirection(stepDirection);
```

Comment:
>((IDebugReversibleEngineProgram160)m_engine) [](http://example.com/codeflow?start=16&length=44)

Same comments here -- test m_program, and handle the engine not implementing it.